addons/website_slides/static/src/js/slides_course_fullscreen_player.js

```diff
@@ -15,309 +15,431 @@ odoo.define('website_slides.fullscreen', function (require) {
+        },
+        //--------------------------------------------------------------------------
+        // Public
+        //--------------------------------------------------------------------------
+        setCurrentSlide: function (slide_id) {
+            // change the UI
+            this.activeTab.removeClass('active');
+            this.activeTabParent.removeClass('active');
+            this.activeTab = this.$('.o_wslides_fs_sidebar_slide_tab[data-id='+slide_id+']');
+            this.activeTabParent = this.$('li[slide_id='+slide_id+']');
+            this.activeTab.addClass('active');
+            this.activeTabParent.addClass('active');
+        },
+        setSlideCompleted: function (slide_id) {
+            // green up the bullet
+            this.$('#check-'+slide_id).replaceWith($('<i class="check-done o_wslides_slide_completed fa fa-check-circle"></i>'));
+        },
+		toggle: function () {
+            this.$('.o_wslides_fs_sidebar').toggleClass('o_wslides_fs_sidebar_hidden');
+		},
+        updateProgressbar: function (channelCompletion) {
+            var completion = Math.min(100, channelCompletion);
+            $('.o_wslides_fs_sidebar_progress_gauge').css('width', completion + "%" );
+            $('.o_wslides_progress_percentage').text(completion);
+        },
+        //--------------------------------------------------------------------------
+        // Handler
+        //--------------------------------------------------------------------------
+        _onClickMiniQuiz: function (ev){
+            var slideIndex = parseInt($(ev.currentTarget).attr('index'));
+            this.trigger_up('click_miniquiz', {slideIndex: slideIndex});
+        },
+		_onClickTab: function (ev) {
+            // trigger_up the event with slide id
+            var slideIndex = parseInt($(ev.currentTarget).attr('index'));
+			this.trigger_up('change_slide', {slideIndex: slideIndex});
+        },
+	});
 
     var Fullscreen = Widget.extend({
+        custom_events: {
+            "toggle_sidebar": '_onToggleSidebar',
+            "change_slide": '_onClickSlideTab',
+            "click_miniquiz": '_onClickMiniquizTab'
+        },
         /**
         * @override
         * @param {Object} el
         * @param {Object} data holding channelId and optionally upload and publish control parameters
         */
-        init: function(el, course_id,slide_id, user_id){
-            this._super.apply(this,arguments);
-            this.courseID = parseInt(course_id, 10);
-            this.slideID = parseInt(slide_id, 10);
-            this.userID = parseInt(user_id, 10);
-            this.course = undefined;
-            this.slide = undefined;
-            this.slides = [];
-            this.nextSlide = undefined;
-            this.previousSlide = undefined;
+        init: function (el, slides, currentSlideIndex){
+            this.slides = slides;
+            this.index = currentSlideIndex;
+            this.slide = this.slides[this.index];
```

Comment:
You should have `this.set('slide', false); this.on('change:slide, this._onchangeSlide)`. The handler `_onchangeSlide` will be called when slide is changed, so in there, you can do something like
```
this.renderContent().then(function(){
  self.header.setTitle();
  if (need to set as done according to type) {
          self.setAsDone();
   }
});
```

You should have a method to find the slide in the slide list from its `slide_id`. Look at https://underscorejs.org/#findIndex, maybe that can help you. (this lib exists in odoo, so use it, this is very usefull)

Reply:
Done